dlt/extract/hints.py

```diff
@@ -341,6 +347,11 @@ def apply_hints(
                     t["file_format"] = file_format
                 else:
                     t.pop("file_format", None)
+            if references is not None:
+                if references:
+                    t["references"] = references
```

Comment:
Hmm yeah I think best would be to replace the reference with same table name, rather than trying to merge the list of columns. Probably that's more obvious.